pwiz_tools/Skyline/Model/Results/SpectrumFilter.cs

```diff
@@ -1,1149 +1,1298 @@
+
+        public IEnumerable<SpectrumFilterPair> FilterPairs { get { return _filterMzValues; } }
+
+        public SpectrumFilter(SrmDocument document, MsDataFileUri msDataFileUri, IFilterInstrumentInfo instrumentInfo, 
+            double? maxObservedIonMobilityValue = null,
+            IRetentionTimePredictor retentionTimePredictor = null, bool firstPass = false, GlobalChromatogramExtractor gce = null)
+        {
+            _fullScan = document.Settings.TransitionSettings.FullScan;
+            _instrument = document.Settings.TransitionSettings.Instrument;
+            _acquisitionMethod = _fullScan.AcquisitionMethod;
+            _ionMobilityFunctionsProvider = instrumentInfo;
+            if (instrumentInfo != null)
+            {
+                _isWatersFile = instrumentInfo.IsWatersFile;
+                _configInfoList = instrumentInfo.ConfigInfoList;
+            }
+            IsFirstPass = firstPass;
+
+            var comparer = PrecursorTextId.PrecursorTextIdComparerInstance;
+            var dictPrecursorMzToFilter = new SortedDictionary<PrecursorTextId, SpectrumFilterPair>(comparer);
+
+            // If we're using bare measured ion mobility values from spectral libraries, go get those now
+            var libraryIonMobilityInfo = document.Settings.PeptideSettings.Prediction.UseLibraryIonMobilityValues
+                ? document.Settings.GetIonMobilities(msDataFileUri)
+                : null;
+            var ionMobilityMax = maxObservedIonMobilityValue ?? 0;
+
+            // TIC and Base peak are meaningless with FAIMS, where we can't know the actual overall ion counts -also can't reliably share times with any ion mobility scheme
+            if (instrumentInfo != null && instrumentInfo.IonMobilityUnits != eIonMobilityUnits.none)
+            {
+                foreach (var pair in document.MoleculePrecursorPairs)
+                {
+                    double windowIM;
+                    var ionMobility = document.Settings.GetIonMobility(
+                        pair.NodePep, pair.NodeGroup, null, libraryIonMobilityInfo, _ionMobilityFunctionsProvider, ionMobilityMax, out windowIM);
+                    _isIonMobilityFiltered = ionMobility.HasIonMobilityValue;
+                    if (_isIonMobilityFiltered)
+                    {
+                        break;
+                    }
+                }
+            }
+
+            if (EnabledMs || EnabledMsMs)
+            {
+                if (EnabledMs)
+                {
+                    _isHighAccMsFilter = !Equals(_fullScan.PrecursorMassAnalyzer,
+                        FullScanMassAnalyzerType.qit);
+
+                    /*
+                     Leaving this here in case we ever decide to fall back to our own BPC+TIC extraction in cases where data
+                     file doesn't have them ready to go, as in mzXML
+                    if (!firstPass && _isIonMobilityFiltered)
```

Comment:
This change looks suspicious. Why did it go from "not faims" to "yes ion mobility filtered"?

Reply:
Agreed. A replace-all goof, I think.